src/lib/Libutil/misc_utils.c

```diff
@@ -725,16 +724,16 @@ replace(char *str, char	*sub, char *repl, char	*retstr)
  * @param[in]	delim  	- delimiter to be searched on str
  * @param[in]   esc 	- escape character to be added if delim found on str
  * @param[out]  retstr 	- delim replaced with and escape character+delim if found.
- * @param[in]   retstrlen 	- length of the output buffer
  *
  * @return void
  */
 
 void
-escape_delimiter(char *str, char *delim, char esc, char *retstr, int retstrlen)
+escape_delimiter(char *str, char *delim, char esc, char *retstr)
 {
 	int	i=0, j=0;
 	int	delim_len;
+	char	rstr[MAXBUFLEN+1];
```

Comment:
If you really want to go the route of creating a buffer on stack then you might as well make it static